tests/test_modules.py

```diff
@@ -620,3 +620,40 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
 
         self.assertEqual(cnt.op_count, 4)
         self.assertTrue(torchdynamo.testing.same(out1, out2))
+
+    def test_torch_function_with_closure(self):
```

Comment:
this has some overlap with `test_simple_torch_function`, but making parts of the code reusable seemed overkill, I'm flexible though